src/targets/npm.ts

```diff
@@ -246,3 +263,95 @@ export class NpmTarget extends BaseTarget {
     this.logger.info('NPM release complete');
   }
 }
+
+/**
+ * Get the latest version for the given package.
+ */
+export async function getLatestVersion(
+  packageName: string,
+  npmConfig: NpmTargetOptions,
+  otp?: NpmPublishOptions['otp']
+): Promise<string | undefined> {
+  const args = ['info', packageName, 'version'];
+  const bin = NPM_BIN;
+
+  try {
+    const response = await withTempFile(filePath => {
+      // Pass OTP if configured
+      const spawnOptions: SpawnOptions = {};
+      spawnOptions.env = { ...process.env };
+      if (otp) {
+        spawnOptions.env.NPM_CONFIG_OTP = otp;
+      }
+      spawnOptions.env[NPM_TOKEN_ENV_VAR] = npmConfig.token;
+      // NOTE(byk): Use npm_config_userconfig instead of --userconfig for yarn compat
+      spawnOptions.env.npm_config_userconfig = filePath;
+      writeFileSync(
+        filePath,
+        `//registry.npmjs.org/:_authToken=\${${NPM_TOKEN_ENV_VAR}}`
+      );
+
+      // Disable output buffering because NPM/Yarn can ask us for one-time passwords
+      return spawnProcess(bin, args, spawnOptions, {});
+    });
+
+    if (!response) {
+      return undefined;
+    }
+
+    return response.toString().trim();
+  } catch {
+    return undefined;
+  }
+}
+/**
+ * Get the tag to use for publishing to npm.
+ */
+export async function getPublishTag(
+  version: string,
+  checkPackageName: string | undefined,
+  npmConfig: NpmTargetOptions,
+  logger: NpmTarget['logger'],
+  otp?: NpmPublishOptions['otp']
+): Promise<string | undefined> {
+  if (isPreviewRelease(version)) {
+    logger.warn('Detected pre-release version for npm package!');
+    logger.warn('Adding tag "next" to not make it "latest" in registry.');
+    return 'next';
+  }
+
+  // If no checkPackageName is given, we return undefined
+  if (!checkPackageName) {
+    return undefined;
+  }
+
+  const latestVersion = await getLatestVersion(
+    checkPackageName,
+    npmConfig,
+    otp
+  );
+  const parsedLatestVersion = latestVersion && parseVersion(latestVersion);
+  const parsedNewVersion = parseVersion(version);
+
+  if (!parsedLatestVersion) {
+    logger.warn(
+      `Could not fetch current version for package ${checkPackageName}`
+    );
+    return undefined;
+  }
+
+  // If we are publishing a version that is older than the currently latest version,
+  // We tag it with "old" instead of "latest"
+  if (
+    parsedNewVersion &&
+    !versionGreaterOrEqualThan(parsedNewVersion, parsedLatestVersion)
+  ) {
+    logger.warn(
+      `Detected older version than currently published version (${latestVersion}) for ${checkPackageName}`
+    );
+    logger.warn('Adding tag "old" to not make it "latest" in registry.');
+    return 'old';
```

Comment:
theoretically (not saying this should happen now), we could assign a `<major>.x` tag here if we decide we want to go this route for multi-major version support.